docs/sdk2/usage/CUSTOM_OBJECT_SYNC.md

```diff
@@ -0,0 +1,257 @@
+
+####  SyncOptions
+
+After the `projectApiRoot` is set up, a `CustomObjectSyncOptions` should be built as follows:
+````java
+// instantiating a CustomObjectSyncOptions
+final CustomObjectSyncOptions customObjectSyncOptions = CustomObjectSyncOptionsBuilder.of(projectApiRoot).build();
+````
+
+`SyncOptions` is an object which provides a place for users to add certain configurations to customize the sync process.
+Available configurations:
+
+##### errorCallback
+A callback that is called whenever an error event occurs during the sync process. Each resource executes its own 
+error-callback. When the sync process of a particular resource runs successfully, it is not triggered. It contains the 
+following context about the error-event:
+
+* sync exception
+* custom object draft from the source
+* custom object of the target project (only provided if an existing custom object could be found)
+* a fake list of update actions, as custom objects API does not provide update actions. [NoopResourceUpdateAction.java](#todo)
+
+````java
+ final Logger logger = LoggerFactory.getLogger(CustomObjectSync.class);
+ final CustomObjectSyncOptions customObjectSyncOptions = CustomObjectSyncOptionsBuilder
+         .of(projectApiRoot)
+         .errorCallback((syncException, draft, customObject, updateActions) -> 
+            logger.error(new SyncException("My customized message"), syncException)).build();
+````
+    
+##### warningCallback
+A callback is called whenever a warning event occurs during the sync process. Each resource executes its own 
+warning-callback. When the sync process of a particular resource runs successfully, it is not triggered. It contains the 
+following context about the warning message:
+
+* sync exception
+* custom object draft from the source 
+* custom object of the target project (only provided if an existing custom object could be found)
+
+````java
+ final Logger logger = LoggerFactory.getLogger(CustomObjectSync.class);
+ final CustomObjectSyncOptions customObjectSyncOptions = CustomObjectSyncOptionsBuilder
+         .of(projectApiRoot)
+         .warningCallback((syncException, draft, customObject) -> 
+            logger.warn(new SyncException("My customized message"), syncException)).build();
+````
+
+##### beforeUpdateCallback
+In theory, `CustomObjectSyncOptions` provides callback before update operation. User can customize their own callback and inject
+into sync options. However, in the actual case, `beforeUpdateCallback`is not triggered in the custom object sync process. When
+the new custom object draft has the same key and container as an existing custom object but different in custom object values, 
+the sync process automatically performs the create/update operation. The value of a corresponding custom object in the target project is overwritten. This approach is different from other resources and no update action is involved.
+
+No example is applicable.
+
+##### beforeCreateCallback
```

Comment:
This callback also applies for updates, right? If so, please mention it here as well.

Reply:
It calls the same API endpoint, but the sync calculates if an update or creation is needed. This callback is only applied when the given customObject doesn't already exist in the project. So, from this perspective it's only in case of custom object creation.